brick-editor.js

```diff
@@ -35,163 +37,487 @@ function addBlocksHTML() { // eslint-disable-line no-unused-vars
 // EVENT HANDLERS
 
 /**
- * Handles button clicks
+ * Called when backspace key is pressed
+ * @returns {undefined}
+ */
+function backspaceHandler() {
+    // if has selected, delete selected
+    var selection = hasSelected();
+    var cursor = getCursor();
+    var buffer = editor.getValue();
+    var ast = recast.parse(buffer);
+    var node = null;
+    if (selection) {
+        node = deleteSelected(ast, selection);
```

Comment:
This function and the branches are hard to understand right now. Could each branch be pulled into a function?